src/Compilers/Core/Portable/PEWriter/RootModuleType.cs

```diff
@@ -18,6 +20,18 @@ namespace Microsoft.Cci
     /// </summary>
     internal class RootModuleType : INamespaceTypeDefinition
     {
+        private IReadOnlyList<IMethodDefinition> _methods = SpecializedCollections.EmptyReadOnlyList<IMethodDefinition>();
```

Comment:
>private IReadOnlyList<IMethodDefinition> _methods = SpecializedCollections.EmptyReadOnlyList<IMethodDefinition>(); [](start = 8, length = 114)

This approach doesn't allow us to assert that SetStaticConstructorBody is never called after GetMethods is called. I think we would like to have an assert like that in place. Consider to follow the approach I suggested earlier:
- we start with null
- SetStaticConstructorBody asserts it is null and assigns one element collection.
- GetMethods assigns an empty collection if null

Reply:
Sorry I missed your suggestion!